### ISSUE TITLE
Hi, I'm the person who mailed 3 days before.

### ISSUE THREAD

**Issue by ssm0414** [initial-h/AlphaZero_Gomoku_MPI#32]
You said 'leave the question here' , so I write the question here.

After I mailed you, I compared with mcts_alphazero.py and mcts_pure.py as far as i could.

I found something difference. In mcts_alphazero, more variables are added(like dirichlet, temperature)

As far as I understood, variables like dirichlet and temperature in mcts_alphazero are devices that can explore more than mcts_pure... 

are there any differences between them? am i understanding right?

**Comment by initial-h** [comment]
yeah, feel free to open issues for any questions.

These variables are for the neural network in alphazero to add noises in the tree before choosing actions, which is in order to make a trade-off between exploration and exploitation and get all kinds of samples. In mcts_pure, there is no network, the probabilities for actions are given uniformly.
https://github.com/initial-h/AlphaZero_Gomoku_MPI/blob/95867cb7e524ebe9c77a926c82091785693c3c0a/mcts_pure.py#L21

**Comment by ssm0414** [comment]
hi again, I have another question.

In your slides, there are sentences ( MCTS for kifu collecting and Deep learning for kifu fitting )

Do they mean Supervised-learning??

If not, what do they mean?

**Comment by initial-h** [comment]
yes. when we have collected data from MCTS, we have state-label pairs (state,probs,value), then the network is trained by supervised learning.

**Comment by ssm0414** [comment]
> yes. when we have collected data from MCTS, we have state-label pairs (state,probs,value), then the network is trained by supervised learning.

Is Mcts_pure.py needed for supervised learning??

**Comment by initial-h** [comment]
no, only the network needs to be trained.

**Comment by ssm0414** [comment]
> > yes. when we have collected data from MCTS, we have state-label pairs (state,probs,value), then the network is trained by supervised learning.
> 
> Is Mcts_pure.py needed for supervised learning??

Sorry, but I don't understand the meaning of Data from MCTS... As I read alphago-zero's paper, it doesn't need any kifu and supervised learning. Do i know wrong?

**Comment by initial-h** [comment]
in the paper 'Mastering the game of Go without human knowledge', the METHODS part introduced the concrete details of 'Supervised learning'.

**Comment by ssm0414** [comment]
Hi again.

I run the train.py because I want to know the train process.
but when I run train.py , there is a phrase "can not find saved model, learn from scratch !"

I want to know what the phrase mean. 

thank you for your reply and take care of yourself

**Comment by initial-h** [comment]
hi， this is because I change the directory's name when I update a 15x15 model, so that it cannot find the trained model and start with a random network.
You can modify it to init_model='model_11_11_5/best_policy.model' in the following line
https://github.com/initial-h/AlphaZero_Gomoku_MPI/blob/95867cb7e524ebe9c77a926c82091785693c3c0a/train.py#L261
it will load the model and continue to train from this model

**Comment by ssm0414** [comment]
I have another problem.
When I run the train.py, I could get values about kl, loss, entropy, explained_var_old and explained_var_new.
I have a question about loss and entropy,
loss value is about 2.8-3.0
entropy value is similar with loss value (2.8-3.0)
Is there any relationship between loss and entropy?

**Comment by initial-h** [comment]
no, I think it's just coincidence
